src/slice/mod.rs

```diff
@@ -67,6 +67,22 @@ pub trait ParallelSlice<T: Sync> {
         }
     }
 
+    /// Returns a parallel iterator over all contiguous array windows of
+    /// length `N`. The windows overlap.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    /// let windows: Vec<_> = [1, 2, 3].par_array_windows().collect();
+    /// assert_eq!(vec![&[1, 2], &[2, 3]], windows);
+    /// ```
+    fn par_array_windows<const N: usize>(&self) -> ArrayWindows<'_, T, N> {
```

Comment:
clearly we're not going to land with `const N: usize` in here, right?

Reply:
Pending stable `min_const_generics`, yes!

Otherwise, how would you do this for arrays? We already have `par_windows` to get slices.

Reply:
This is now `build.rs` detected.